config/crd/bases/oadp.openshift.io_nonadmindownloadrequests.yaml

```diff
@@ -0,0 +1,186 @@
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.14.0
+  name: nonadmindownloadrequests.oadp.openshift.io
+spec:
+  group: oadp.openshift.io
+  names:
+    kind: NonAdminDownloadRequest
+    listKind: NonAdminDownloadRequestList
+    plural: nonadmindownloadrequests
+    singular: nonadmindownloadrequest
+  scope: Namespaced
+  versions:
+  - name: v1alpha1
+    schema:
+      openAPIV3Schema:
+        description: NonAdminDownloadRequest is the Schema for the nonadmindownloadrequests
+          API.
+        properties:
+          apiVersion:
+            description: |-
+              APIVersion defines the versioned schema of this representation of an object.
+              Servers should convert recognized schemas to the latest internal value, and
+              may reject unrecognized values.
+              More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources
+            type: string
+          kind:
+            description: |-
+              Kind is a string value representing the REST resource this object represents.
+              Servers may infer this from the endpoint the client submits requests to.
+              Cannot be updated.
+              In CamelCase.
+              More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: |-
+              NonAdminDownloadRequestSpec defines the desired state of NonAdminDownloadRequest.
+              Mirrors velero DownloadRequestSpec to allow non admins to download information for a non admin backup/restore
+            properties:
+              target:
+                description: Target is what to download (e.g. logs for a backup).
+                properties:
+                  kind:
+                    description: Kind is the type of file to download.
+                    enum:
+                    - BackupLog
+                    - BackupContents
+                    - BackupVolumeSnapshots
+                    - BackupItemOperations
+                    - BackupResourceList
+                    - BackupResults
+                    - RestoreLog
+                    - RestoreResults
+                    - RestoreResourceList
+                    - RestoreItemOperations
+                    - CSIBackupVolumeSnapshots
```

Comment:
dumb boss question, does the downloadrequest do any checking to ensure that snapshots are tied to the backup?  Maybe because of the object storage directory structure there is no chance of pulling a snapshot or vsc that is NOT tied to the backup / resture.  /me just checking the privacy / security is bullet proof :)

Reply:
@weshayutin This is VolumeSnapshot CR metadata, not kopia volume content.

Reply:
It's a file uploaded as part of the backup info, so it should be in the directory in obj storage for the specific backup.

Reply:
```
func (l *ObjectStoreLayout) getCSIVolumeSnapshotKey(backup string) string {
        return path.Join(l.subdirs["backups"], backup, fmt.Sprintf("%s-csi-volumesnapshots.json.gz", backup))
}
```
So the download URL here will be bucket-name/backups/backup-name/backup-name-csi-olumesnapshots.json.gz

Reply:
Also relevant is the TODO in https://github.com/migtools/oadp-non-admin/issues/219
"Test that signed urls from download requests cannot be used to access other paths in the bucket"

Reply:
That is out of scope for FC. we're limiting to NAbsl